client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -0,0 +1,374 @@
+				...getEventPropsByDesign( _selectedDesign ),
+				...( positionIndex >= 0 && { position_index: positionIndex } ),
+			} );
+
+			handleSubmit( {
+				selectedDesign: _selectedDesign,
+				selectedSiteCategory: categorization.selection,
+			} );
+		}
+	}
+
+	function previewDesign( _selectedDesign: Design, positionIndex?: number ) {
+		recordTracksEvent( 'calypso_signup_design_preview_select', {
+			...getEventPropsByDesign( _selectedDesign ),
+			...( positionIndex && { position_index: positionIndex } ),
+		} );
+
+		setSelectedDesign( _selectedDesign );
+		setIsPreviewingDesign( true );
+	}
+
+	function upgradePlan() {
+		if ( ! isEnabled( 'signup/seller-upgrade-modal' ) ) {
+			return goToCheckout();
+		}
+
+		recordTracksEvent( 'calypso_signup_design_upgrade_modal_show', {
+			theme: selectedDesign?.slug,
+		} );
+		setShowUpgradeModal( true );
+	}
+
+	function goToCheckout() {
+		if ( ! isEnabled( 'signup/design-picker-premium-themes-checkout' ) ) {
+			return null;
+		}
+
+		if ( isEnabled( 'signup/seller-upgrade-modal' ) ) {
+			recordTracksEvent( 'calypso_signup_design_upgrade_modal_checkout_button_click', {
+				theme: selectedDesign?.slug,
+			} );
+		}
+
+		const plan = isEligibleForProPlan && isEnabled( 'plans/pro-plan' ) ? 'pro' : 'premium';
+
+		if ( siteSlugOrId ) {
+			const params = new URLSearchParams();
+			params.append( 'redirect_to', window.location.href.replace( window.location.origin, '' ) );
+
+			// The theme upsell link does not work with siteId and requires a siteSlug.
+			// See https://github.com/Automattic/wp-calypso/pull/64899
+			window.location.href = `/checkout/${ encodeURIComponent(
+				siteSlug || urlToSlug( site?.URL || '' ) || ''
+			) }/${ plan }?${ params.toString() }`;
+		}
+	}
+
+	function recordStepContainerTracksEvent( eventName: string ) {
+		const tracksProps = {
+			step: 'design-step',
+			intent: intent,
+		};
+
+		recordTracksEvent( eventName, tracksProps );
+	}
+
+	const handleBackClick = () => {
+		if ( isPreviewingDesign ) {
+			setSelectedDesign( undefined );
+			setIsPreviewingDesign( false );
+			return;
+		}
+
+		goBack();
+	};
+
+	// Track scroll event to make sure people are scrolling on mobile.
+	useTrackScrollPageFromTop( isMobile && ! isPreviewingDesign, flow || '', STEP_NAME );
+
+	// Make sure people is at the top when entering/leaving preview mode.
+	useEffect( () => {
+		window.scrollTo( { top: 0 } );
+	}, [ isPreviewingDesign ] );
+
+	// Don't render until we've fetched the designs from the backend.
+	if ( ! site || isLoadingDesigns ) {
+		return null;
+	}
+
+	if ( selectedDesign && isPreviewingDesign ) {
+		const designTitle = selectedDesign.title;
+		const shouldUpgrade = selectedDesign.is_premium && ! isPremiumThemeAvailable;
+		const previewUrl = getDesignPreviewUrl( selectedDesign, {
+			language: locale,
+			// If the user fills out the site title with write intent, we show it on the design preview
+			siteTitle: intent === 'write' ? siteTitle : undefined,
```

Comment:
I think @arthur791004 recently updated the parameter signature, see: https://github.com/Automattic/wp-calypso/pull/65808/files#diff-7f3668fb59958a4a910fb057f3b023e125035331ee35488b3d8ab614a2d0d4ecR68

Reply:
ported his changes across 👍